src/lib/mode_handler.js

```diff
@@ -53,7 +53,10 @@ var ModeHandler = function(mode, DrawContext) {
 
   return {
     render: mode.render || function(geojson) {return geojson; },
-    stop: mode.stop || function() {},
+    stop: function() {
+      DrawContext.store.clearSelected();
```

Comment:
Lets do this after `mode.stop`, just in case a mode wants to do something with the selected features on end.